plugins/destination/bigquery/client/read.go

```diff
@@ -97,20 +36,153 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 	for {
-		values := c.createResultsArray(table)
-		v := make([]any, len(values))
+		values := make([]bigquery.Value, len(table.Columns))
 		err := it.Next(&values)
 		if err == iterator.Done {
 			break
 		}
 		if err != nil {
 			return fmt.Errorf("failed to read from table %s: %w", table.Name, err)
 		}
+		rb := array.NewRecordBuilder(memory.DefaultAllocator, arrowSchema)
 		for i := range values {
-			v[i] = values[i]
+			readItem(values, rb, i)
 		}
-		res <- v
+		res <- rb.NewRecord()
 	}
 	return nil
 }
+
+func readItem(values []bigquery.Value, rb *array.RecordBuilder, i int) error {
+	fieldBuilder := rb.Field(i)
+	value := values[i]
+	return appendValue(fieldBuilder, value)
+}
+
+func appendValue(bldr array.Builder, value any) error {
+	if value == nil {
+		bldr.AppendNull()
+		return nil
+	}
+	switch arr := bldr.(type) {
+	case *array.StructBuilder:
+		m := value.([]bigquery.Value)
+		arr.Append(true)
+		for f := 0; f < arr.NumField(); f++ {
+			fieldBldr := arr.FieldBuilder(f)
+			if err := appendValue(fieldBldr, m[f]); err != nil {
+				return err
+			}
+		}
+		return nil
+	case array.ListLikeBuilder:
+		lst := value.([]bigquery.Value)
+		if lst == nil {
+			arr.AppendNull()
+			return nil
+		}
+		arr.Append(true)
+		valBuilder := arr.ValueBuilder()
+		for _, v := range lst {
+			if err := appendValue(valBuilder, v); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	switch rbv := bldr.(type) {
+	case *array.Time32Builder:
+		t := value.(civil.Time)
+		unit := bldr.Type().(*arrow.Time32Type).Unit
+		ts := stringForTime(t, unit)
+		t32, err := arrow.Time32FromString(ts, unit)
+		if err != nil {
+			return err
+		}
+		rbv.Append(t32)
+		return nil
+	case *array.Time64Builder:
+		t := value.(civil.Time)
+		unit := bldr.Type().(*arrow.Time64Type).Unit
+		ts := stringForTime(t, unit)
+		t64, err := arrow.Time64FromString(ts, unit)
+		if err != nil {
+			return err
+		}
+		rbv.Append(t64)
+		return nil
+	case *array.DayTimeIntervalBuilder:
+		t := value.([]bigquery.Value)
+		rbv.Append(arrow.DayTimeInterval{
+			Days:         int32(t[0].(int64)),
+			Milliseconds: int32(t[1].(int64)),
+		})
+		return nil
+	case *array.MonthDayNanoIntervalBuilder:
+		t := value.([]bigquery.Value)
+		rbv.Append(arrow.MonthDayNanoInterval{
+			Months:      int32(t[0].(int64)),
+			Days:        int32(t[1].(int64)),
+			Nanoseconds: t[2].(int64),
+		})
+		return nil
+	case *array.MonthIntervalBuilder:
+		t := value.([]bigquery.Value)
+		rbv.Append(arrow.MonthInterval(t[0].(int64)))
+		return nil
+	case *array.TimestampBuilder:
+		unit := bldr.Type().(*arrow.TimestampType).Unit
+		switch unit {
+		case arrow.Nanosecond:
+			v := value.([]bigquery.Value)
+			t := v[0].(time.Time)
+			nano := v[1].(int64)
+			nanoTime := t.Add(time.Duration(nano) * time.Nanosecond)
+			ts, err := arrow.TimestampFromString(nanoTime.Format(time.RFC3339Nano), arrow.Nanosecond)
+			if err != nil {
+				return fmt.Errorf("failed to call arrow.TimestampFromString: %w", err)
```

Comment:
```suggestion
				return err
```
to match the next case

Reply:
Thanks!